src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,186 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { pathExists } from 'fs-extra';
+import * as path from 'path';
+import * as semver from 'semver';
+import { MessageItem, OpenDialogOptions, workspace } from 'vscode';
+import { UserCancelledError } from 'vscode-azureextensionui';
+import { Platform, TemplateFilter } from "../../constants";
+import { ext } from '../../extensionVariables';
+import { validateFuncCoreToolsInstalled } from '../../funcCoreTools/validateFuncCoreToolsInstalled';
+import { localize } from "../../localize";
+import { getFuncExtensionSetting, updateWorkspaceSetting } from '../../ProjectSettings';
+import { cpUtils } from "../../utils/cpUtils";
+import { funcHostTaskId } from "./IProjectCreator";
+import { ScriptProjectCreatorBase } from './ScriptProjectCreatorBase';
+
+export class PythonProjectCreator extends ScriptProjectCreatorBase {
+    public readonly templateFilter: TemplateFilter = TemplateFilter.Verified;
+    private readonly pythonVenvPathSetting: string = 'python.venvRelativePath';
+    private pythonAlias: string;
+    public getLaunchJson(): {} {
+        // currently the Python extension launches Windows with type: python and MacOS with type: pythonExperimental
+        const launchType: string = process.platform === Platform.Windows ? 'python' : 'pythonExperimental';
+        return {
+            version: '0.2.0',
+            configurations: [
+                {
+                    name: localize('azFunc.attachToJavaScriptFunc', 'Attach to Python Functions'),
+                    type: launchType,
+                    request: 'attach',
+                    port: 9091,
+                    host: 'localhost',
+                    preLaunchTask: funcHostTaskId
+                }
+            ]
+        };
+    }
+
+    public async addNonVSCodeFiles(): Promise<void> {
+        // need a custom message for warning as this is a dependency
+        if (await validateFuncCoreToolsInstalled(true)) {
+            await this.validatePythonVersion();
+            await this.ensureVirtualEnviornment();
+            await this.createPythonProject();
+        } else {
+            throw new UserCancelledError();
```

Comment:
I don't quite follow why this case should be a UserCancelledError?

Reply:
The prompt for the Functions installation is a forced modal prompt.  But if they hit Cancel, it wasn't being handled as a `UserCancelledError`.

Reply:
Refer to this: https://github.com/Microsoft/vscode-azurefunctions/blob/master/src/commands/pickFuncProcess.ts#L19

Reply:
>  But if they hit Cancel, it wasn't being handled as a UserCancelledError.

Then we should fix that?

I honestly don't get why pickFuncProcess has a `UserCancelledError` either (Yes I realize the irony since I'm the one that wrote that code haha)

Reply:
I think it's because originally when we wrote the function, we didn't account for having a `forcePrompt` property.  So at that time, we didn't want a cancel to throw an error and cancel the entire operation.  I guess the real fix would be to throw a `UserCancelledError` only in the forcePrompt scenario (which I will fix)